mmeds/tools/lefse.py

```diff
@@ -79,13 +95,15 @@ def features(self):
                                        output=self.get_file('features_all')))
 
     def setup_analysis(self, summary=False):
-        self.subclass = 'subclass' in self.doc.reads_type
-        self.subjects = 'subjects' in self.doc.reads_type
         self.set_stage(0)
-        self.format_input()
+        self.extract_qiime2_feature_table('taxa_collapsed_relative_table')
         self.set_stage(1)
-        self.lefse()
+        self.preprocess_feature_table()
+        self.source_activate('lefse')
+        self.format_input()
         self.set_stage(2)
```

Comment:
I like how the code is organized into function calls here. set_stage() isn't a very meaningful name to me at least though. Not implying you should go through the PR renaming stuff, just pointing out.

Reply:
yeah, this is all still david's framework. I kind of like set_stage, since basically all it does is add `echo "MMEDS STAGE #"` to the jobfile. There's a "stage" variable as well in the tool class but it doesn't really do much. Just kind of a structural thing tbh